internal/config/flags.go

```diff
@@ -0,0 +1,19 @@
+package config
+
+import "context"
+
+type GlobalFlags struct {
+	// SkipCatalogUpdate global flag used to skip catalog update and dirty check.
+	SkipCatalogUpdate bool
+}
+
+type flagKey struct{}
+
+func ToFlagContext(parent context.Context, flags *GlobalFlags) context.Context {
+	return context.WithValue(parent, flagKey{}, flags)
+}
+
+func FromFlagContext(ctx context.Context) *GlobalFlags {
```

Comment:
Can we rename this to `FlagsFromContext`? I don't know. I know you are using the name I used as an example yesterday, and I am not 100% convinced its a better name, and generally speaking I don't mind naming too much.

Feel free to disregard this comment but if you think it is an improvement hit F2! :)